chessboard/skoved/chessboard.go

```diff
@@ -0,0 +1,59 @@
+package chessboard
+
+// Declare a type named Rank which stores if a square is occupied by a piece - this will be a slice of bools
+type Rank []bool
+
+// Declare a type named Chessboard contains a map of eight Ranks, accessed with values from "A" to "H"
+type Chessboard map[string]Rank
+
+// validateFile checks if an int is a valid chessboard file (1-8)
+func validateFile(file int) bool {
+	return file >= 1 && file <= 8
+}
+
+// CountInRank returns how many squares are occupied in the chessboard,
+// within the given rank
+func CountInRank(cb Chessboard, rank string) int {
+	numOccupied := 0
+	row, exists := cb[rank]
+	if exists {
+		for _, square := range row {
+			if square {
+				numOccupied++
+			}
+		}
+	}
+	return numOccupied
+}
+
+// CountInFile returns how many squares are occupied in the chessboard,
+// within the given file
+func CountInFile(cb Chessboard, file int) int {
+	numOccupied := 0
+	if validateFile(file) {
+		for _, rank := range cb {
+			if rank[file-1] {
+				numOccupied++
+			}
+		}
+	}
+	return numOccupied
+}
+
+// CountAll should count how many squares are present in the chessboard
+func CountAll(cb Chessboard) int {
```

Comment:
This is where using a rank of a fixed size helps a lot. You could replace everything with something like `return len(cb) * rankSize`.

Reply:
I changed Rank to have a fixed size and updated this function like you suggested